### PR TITLE
Fix bug in semantic analysis with overriding name in WITH .. ORDER BY

### PR DESCRIPTION
**Pull request by sherfert**
Fixes a bug in semantic analysis that would yield an error on such a
query: `MATCH (n) RETURN n.prop AS n ORDER BY n + 2`

This error was previously hidden by rewriters.

### CODE REVIEW

**Comment by sherfert** [comment]
@disolovyov I think this should fix the bug you encountered.

**Comment by disolovyov** [comment]
@dwitry Could you take a look, please? I might not find time this week.

**Comment by dwitry** [comment]
Thank you @sherfert, can confirm, this actually fixes the issue I've reported on Friday. Waiting for release so we could upgrade Cypher for Gremlin.

File: ast/src/main/scala/org/opencypher/v9_0/ast/Clause.scala [hunk 1]
```diff
@@ -658,11 +668,17 @@ sealed trait ProjectionClause extends HorizonClause {
...
+      // Since the current scoping system does not allow that in a nice way, we run the semantic check for these two twice
       // In the first run we construct a scope with all necessary variables defined to find errors in these subclauses
       val specialReturnItems = createSpecialReturnItems(previousScope, s)
-      val specialStateForOrderByAndWhere = specialReturnItems.declareVariables(previousScope)(s).state
-      val orderByAndWhereResult = (orderBy.semanticCheck chain checkSkip chain checkLimit chain where.semanticCheck) (specialStateForOrderByAndWhere)
+      val specialCheckForOrderByAndWhere =
+        specialReturnItems.declareVariables(previousScope) chain
+        orderBy.semanticCheck chain
+        checkSkip chain
+        checkLimit chain where.
```
Comment (OliviaYtterbrink) [inline ast/src/main/scala/org/opencypher/v9_0/ast/Clause.scala]: Easier to read, but a weird line break here, would prefer `where.semanticCheck` on one line.

**Pull request merged**